src/components/PopupWithForm.js

```diff
@@ -2,7 +2,7 @@ import Popup from "./Popup.js";
 
 export default class PopupWithForm extends Popup {
   constructor(popupSelector, handleFormSubmit) {
-    super({ popupSelector });
+    super(popupSelector);
     this._popupForm = this._popupElement.querySelector(".modal__form");
     this._handleFormSubmit = handleFormSubmit;
     this._inputValues = [...this._popupForm.querySelectorAll(".modal__input")];
```

Comment:
`this._popup = document.querySelector(popupSelector);` is not needed in line 9

* **No** need to search the **modal** again in the inheritors of class `Popup` because the parent has already found it in its constructor as `this._popupElement`. It’s available in any child.

Reply:
Removed and updated instances to this._popupElement instead of this._popup.